src/util/thread_metrics.rs

```diff
@@ -78,62 +78,90 @@ impl Collector for ThreadsColletcor {
     }
 }
 
-fn get_threads(pid: pid_t) -> Result<Vec<(pid_t, String)>> {
-    let mut pairs = Vec::new();
-
+fn get_thread_ids(pid: pid_t) -> Result<Vec<pid_t>> {
+    let mut tids = Vec::new();
     let dirs = try!(fs::read_dir(format!("/proc/{}/task", pid)));
     for task in dirs {
         let tid = match try!(task).file_name().to_str() {
             Some(tid) => try!(tid.parse().map_err(|e| to_err(format!("{}", e)))),
-            None => return Err(to_err(format!("fail to read {} task", pid))),
+            None => {
+                error!("fail to read task of {}", pid);
+                continue;
+            }
         };
-
-        let mut tname = String::new();
-        try!(fs::File::open(format!("/proc/{}/task/{}/comm", pid, tid))
-            .and_then(|mut f| f.read_to_string(&mut tname)));
-
-        pairs.push((tid, sanitize_thread_name(tid, tname)));
+        tids.push(tid);
     }
 
-    Ok(pairs)
+    Ok(tids)
 }
 
-fn sanitize_thread_name(tid: pid_t, tname: String) -> String {
-    let mut san = String::with_capacity(tname.len());
-    for c in tname.chars() {
+fn get_thread_name(tid: pid_t, stat: &str) -> (String, usize) {
+    let mut space_cnt = 0;
+    let mut open_brkt = 0;
+    let mut name = String::with_capacity(THD_NAME_LEN);
+    for c in stat.chars() {
         match c {
+            // TODO: properly handle `)a(`.
```

Comment:
what if the name is ")a S"? A better way to handle this is treat substring between first '(' and last ')' as the thread's name.